src/chain.jl

```diff
@@ -0,0 +1,83 @@
+const Transform = Union{Assertion, Filter, Imputor}
+
+"""
+    Chain
+
+Runs multiple `Assertions`, `Filter` or `Imputor`s on the same data in the order they're
+provided.
+
+# Fields
+* `transforms::Vector{Union{Assertion, Filter, Imputor}}`
+"""
+struct Chain
+    transforms::Vector{Transform}
```

Comment:
```suggestion
struct Chain{T<:NTuple{Transform}}
    transforms::T
```

using a Tuple so the compiler can specialize on this?